configure.ac

```diff
@@ -1202,60 +1178,29 @@ AS_IF([$have_libbpf], [
 PKG_CHECK_MODULES([libelf], [libelf], [have_libelf=true], [have_libelf=false])
 AC_SUBST(HAVE_LIBELF, [$have_libelf])
 
-dnl Auto-discover clang if not provided by the user
-if test -z "$PMDABPF_CLANG"; then
-    AC_PATH_PROG(PMDABPF_CLANG, clang)
-fi
-pmdabpf_clang=$PMDABPF_CLANG
-AC_SUBST(pmdabpf_clang)
-
-dnl Auto-discover llvm-strip if not provided by the user
-if test -z "$PMDABPF_LLVM_STRIP"; then
-    AC_PATH_PROG(PMDABPF_LLVM_STRIP, llvm-strip)
-fi
-pmdabpf_llvm_strip=$PMDABPF_LLVM_STRIP
-AC_SUBST(pmdabpf_llvm_strip)
+test -z "$CLANG" && AC_PATH_PROG(CLANG, clang)
+clang=$CLANG
+AC_SUBST(clang)
 
-dnl Build bpftool if not provided by the user
-build_bpftool=false
-if test -z "$PMDABPF_BPFTOOL"; then
-    if $have_libelf; then
-        PMDABPF_BPFTOOL=`realpath vendor/github.com/libbpf/bpftool/src/bpftool`
-        AC_MSG_NOTICE([using bpftool at $PMDABPF_BPFTOOL])
-        build_bpftool=true
-    else
-        AC_MSG_NOTICE([not building bpftool because libelf was not found, please use --with-pmdabpf-bpftool to specify bpftool])
-    fi
-else
-    AC_MSG_NOTICE([using bpftool at $PMDABPF_BPFTOOL])
-fi
-pmdabpf_bpftool=$PMDABPF_BPFTOOL
-AC_SUBST(pmdabpf_bpftool)
+test -z "$LLVM_STRIP" && AC_PATH_PROG(LLVM_STRIP, llvm-strip)
+llvm_strip=$LLVM_STRIP
+AC_SUBST(llvm_strip)
 
 dnl Get BPF CO-RE arch identifier
 AC_MSG_CHECKING([BPF CO-RE architecture identifier])
 pmdabpf_arch=`uname -m | sed 's/x86_64/x86/' | sed 's/aarch64/arm64/' | sed 's/ppc64le/powerpc/' | sed 's/mips.*/mips/'`
 AC_SUBST(pmdabpf_arch)
 AC_MSG_RESULT($pmdabpf_arch)
 
-dnl Generate vmlinux.h if PMDABPF_BTF is provided by the user,
-dnl otherwise fall back to vendored vmlinux.h
-pmdabpf_vmlinuxh=""
-if test -n "$PMDABPF_BTF"; then
-    pmdabpf_vmlinuxh="`pwd`/src/pmdas/bpf/vmlinux.h"
-    AC_MSG_NOTICE([generating $pmdabpf_vmlinuxh from $PMDABPF_BTF])
-    $PMDABPF_BPFTOOL btf dump file "$PMDABPF_BTF" format c > $pmdabpf_vmlinuxh
+AC_MSG_CHECKING([for vmlinux.h])
+pmdabpf_vmlinuxh="`pwd`/vendor/github.com/iovisor/bcc/libbpf-tools/$pmdabpf_arch/vmlinux.h"
+if test -f "$pmdabpf_vmlinuxh"; then
+    AC_MSG_RESULT($pmdabpf_vmlinuxh)
 else
-    pmdabpf_arch_vmlinux="`pwd`/vendor/github.com/iovisor/bcc/libbpf-tools/$pmdabpf_arch/vmlinux.h"
-    if test -f "$pmdabpf_arch_vmlinux"; then
-	pmdabpf_vmlinuxh="$pmdabpf_arch_vmlinux"
-    else
-	AC_MSG_NOTICE([vmlinux.h for architecture $pmdabpf_arch is not vendored, please use --with-pmdabpf-btf to generate a vmlinux.h for $pmdabpf_arch])
-    fi
+    AC_MSG_RESULT(not found);
```

Comment:
Seems like a useful notice to mention that it's vendored or arch-specific. Something like `AC_MSG_RESULT(not found for arch $pmdabpf_arch)`

Reply:
Good point!
I've updated the error message.